Add Jörmungandr binary decoders for legacy funds

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#779 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] I have force the legacy API to use only 'Mainnet discrimination (even on the Testnet!). Addresses in the genesis file will be Mainnet addresses, so we need the ability to recognize them even if running on testnet.

- [x] I have started a first implementation of binary decoder for the legacy funds. The format for legacy funds is different from the format used for standard genesis fund. The message payload contains only a list of txout (address, value) with no associated inputs? It is still unclear to me what these inputs should be and how we could realistically handle these legacy funds

KtorZ committed:
force legacy API to use Mainnet discrimination, even if the underlying network is connected to testnet

This is because addresses in the genesis file will be mainnet addresses, captured from a snapshot of the current mainnet.

KtorZ committed:
implement first draft of binary decoder for legacy UTxO

The format for legacy funds is different from the format used for
standard genesis fund. The message payload contains only a list of
txout (address, value) with no associated inputs? It is still
unclear to me what these inputs should be and how we could realistically
handle these legacy funds

Anviking commented on lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs:
```diff
@@ -369,6 +370,28 @@ getTransaction n = label "getTransaction" $ do
+getLegacyTransaction :: Int -> Get Tx
+getLegacyTransaction size = do
+    bytes <- lookAhead (getLazyByteString $ fromIntegral size)
+    let tag = runPut (putWord8 (messageTypeTag MsgTypeLegacyUTxO))
```
For the record, had the same thought here as said https://github.com/input-output-hk/cardano-wallet/pull/932#discussion_r340136412, with proposed change in #937.

jonathanknowles committed:
Revise formatting slightly to meet our coding standard.

https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters

jonathanknowles committed:
Use an empty set of inputs when getting a legacy transaction.
